test/unit/trace-test.js

```diff
@@ -247,14 +251,434 @@ test('test trace serialization, three spans', (t) => {
+                        value: carrier[CARRIER_CORRELATION_ID],
+                        scope: CORR_ID_SCOPE_INTERACTION,
+                    }],
+                }],
+            }],
+        }],
+    });
+    t.end();
+});
+
+test('test mixed references, primary parent childOf', (t) => {
+    const recorder = new ListRecorder();
+    const tracer = new APMTracer({ recorder: recorder });
+
+    const rootSpan = tracer.startSpan('root');
+    const childSpan1 = tracer.startSpan('child1', { childOf: rootSpan });
+    const childSpan11 = tracer.startSpan('child11', { childOf: childSpan1 });
+    const childSpan2 = tracer.startSpan('child2', { childOf: rootSpan });
+    const childSpan22 = tracer.startSpan('child22', { childOf: childSpan2 });
+
+    childSpan1.finish();
+    childSpan22.finish();
+    childSpan2.finish();
+    rootSpan.finish();
+
+    const multipleReferencesSpan = tracer.startSpan('follows', {
+        references: [
+            new Constants.Reference(Constants.REFERENCE_FOLLOWS_FROM, childSpan11),
+            new Constants.Reference(Constants.REFERENCE_CHILD_OF, childSpan22.context()),
+        ]});
+
+    multipleReferencesSpan.finish();
+    childSpan11.finish();
+
+    t.deepEqual(recorder.getTraces().length, 1);
+    t.deepEquals(recorder.getTraces()[0], {
+        traceId: rootSpan.getTraceId(),
+        fragmentId: rootSpan.getId(),
+        timestamp: rootSpan.getStartTime() * 1000,
+        nodes: [{
+            type: NODE_TYPE_COMPONENT,
+            uri: null,
+            timestamp: rootSpan.getStartTime() * 1000,
+            operation: rootSpan.getOperationName(),
+            duration: rootSpan.getDuration() * 1000,
+            nodes: [{
+                type: NODE_TYPE_COMPONENT,
+                uri: null,
+                timestamp: childSpan1.getStartTime() * 1000,
+                operation: childSpan1.getOperationName(),
+                duration: childSpan1.getDuration() * 1000,
+                nodes: [{
+                    type: NODE_TYPE_COMPONENT,
+                    uri: null,
+                    timestamp: childSpan11.getStartTime() * 1000,
+                    operation: childSpan11.getOperationName(),
+                    duration: childSpan11.getDuration() * 1000,
+                }],
+            }, {
+                type: NODE_TYPE_COMPONENT,
+                uri: null,
+                timestamp: childSpan2.getStartTime() * 1000,
+                operation: childSpan2.getOperationName(),
+                duration: childSpan2.getDuration() * 1000,
+                nodes: [{
+                    type: NODE_TYPE_COMPONENT,
+                    uri: null,
+                    timestamp: childSpan22.getStartTime() * 1000,
+                    operation: childSpan22.getOperationName(),
+                    duration: childSpan22.getDuration() * 1000,
+                    nodes:[{
+                        type: NODE_TYPE_COMPONENT,
```

Comment:
This is the think I need to update, in case of primary reference I need to add other corr. ids.